pkg/network/server.go

```diff
@@ -836,7 +863,7 @@ func (s *Server) broadcastTxHashes(hs []util.Uint256) {
 	// We need to filter out non-relaying nodes, so plain broadcast
 	// functions don't fit here.
 	s.iteratePeersWithSendMsg(msg, Peer.EnqueuePacket, func(p Peer) bool {
-		return p.Handshaked() && p.Version().Relay
+		return p.IsFullNode()
 	})
```

Comment:
How about this?
```go
 	s.iteratePeersWithSendMsg(msg, Peer.EnqueuePacket, (*Peer).IsFullNode)
```

Reply:
Why pointer? Couldn't it be just
```
s.iteratePeersWithSendMsg(msg, Peer.EnqueuePacket, Peer.IsFullNode)
```
?

Reply:
We have `IsFullNode` defined on pointer and we need some function that accepts pointer to `Peer`, so IIRC that's the right syntax. But I may be wrong, of course.

Reply:
It is possible when we have method defined on struct (https://golang.org/ref/spec#Method_expressions), but Peer is an interface, and `*TCPPeer` implements it (not `TCPPeer`), so IsFullNode will be invoked on pointer in case of 
```
s.iteratePeersWithSendMsg(msg, Peer.EnqueuePacket, Peer.IsFullNode)
```

Reply:
Yep, an intermediate interface changes the picture.